### PR TITLE
Fix picker item width overflow

### LINKED ISSUE #352: Characters overflow past picker border
When items in the picker are longer than the picker width, they break the border and overflow:

Screenshot_2021-06-23_13-18-43

### PR DESCRIPTION
**Pull request by sudormrfbin**
Fixes #352

### CODE REVIEW

**Review by CBenoit — APPROVED** [review]
Looks good to me! :smile:

File: helix-term/src/ui/picker.rs [hunk 1]
```diff
@@ -293,7 +293,7 @@ impl<T: 'static> Component for Picker<T> {
                 inner.x + 3,
                 inner.y + 2 + i as u16,
                 (self.format_fn)(option),
-                inner.width as usize - 1,
+                inner.width as usize - 3, // account for the " > "
```
Comment (archseer) [inline helix-term/src/ui/picker.rs:296]: Should be a `.saturating_sub` so we don't panic if the width is really small

**Review by CBenoit — CHANGES_REQUESTED** [review]
I'll remove my approval just to not merge accidentally until the code is updated to use `saturating_sub`.

File: helix-term/src/ui/picker.rs (hunk 1 above)
Comment (sudormrfbin) [reply helix-term/src/ui/picker.rs:296]: Done 👍🏾

**Review by archseer — APPROVED** [review]

**Comment by pickfire** [comment]
Off by one issue

Screenshot_20210624_114317

**Comment by sudormrfbin** [comment]
@pickfire could you check if #369 solves the one off problem ?

**Comment by pickfire** [comment]
I can't reproduce this problem now, I think it's fixed.

**Pull request merged**